internal/app/service.go

```diff
@@ -430,7 +429,7 @@ func (svc *Service) Initialize() error {
 		secretProvider := bootstrapContainer.SecretProviderFrom(svc.dic.Get)
 		credentials, err := secretProvider.GetSecrets(svc.config.Database.Type)
 		if err != nil {
-			return fmt.Errorf("unable to set RedisStreams password from DB credentials")
+			return fmt.Errorf("unable to set RedisStreams password from DB credentials: %s", err.Error())
```

Comment:
I thought we are supposed to use %w now.

Reply:
Nope, not sure what %w is. It is a string returned from err.Error() so %s is most appropriate. Some folks us %v which will work also, but I like to be more explicate.

Reply:
Also %v leaves it up to reflection to figure our the best way to convert the value to a string. So when it is a string why not skip the reflection and use %s...

Reply:
```suggestion
			return fmt.Errorf("unable to set RedisStreams password from DB credentials: %w", err)
```

Reply:
Ok, just looked up `%w`. It is specifically for when wrapping errors . Good to know. :-) THX!
```
Wrapping an error with %w makes it available to errors.Is and errors.As:

err := fmt.Errorf("access denied: %w", ErrPermission)
...
if errors.Is(err, ErrPermission) ...
```

Reply:
in this case, not sure we want to be wrapping error details? just have some concerned about there is credential leakage potentially if shown wrapping error? @lenny-intel  and @bnevis-i

Reply:
Good point. If that's the case, we shouldn't propagate that error at all.

Reply:
Nope, it isn't leaking the credential  that it couldn't get from the SecretStore. ;-)
The error give details on why if could find the credentials.

Reply:
if that is the case, then it is ok; i was just worries the error was wrapped from vault or redis's server's error which might have some sensitive infor there. ok if not

Reply:
@bnevis-i , you goo with this explanation of why it isn't a creds leak?

Reply:
Yes.